mautrix_facebook/portal.py

```diff
@@ -1027,6 +1040,18 @@ async def _send_post_backfill_dummy(
             fb_txn_id=None,
         ).insert()
 
+        await self.main_intent.send_state_event(
+            self.mxid,
+            event_type=HistorySyncMarkerState,
+            # Use the current timestamp as the state key as that is guaranteed to be unique
+            # per-batch.
+            state_key=str(time.time()),
```

Comment:
I was just trying to follow the current MSC. See the [Adding Marker Events](https://github.com/matrix-org/matrix-spec-proposals/blob/matthew/msc2716/proposals/2716-batch-send-historical-messages.md#adding-marker-events) section which includes the following bullet point:

> * Also instead of overwriting the same generic `state_key: ""` over and over, **the expected behavior is send each "marker" event with a unique `state_key`**. This way all of the "markers" are discoverable in the current state without us having to go through the chain of previous state to figure it all out. This also avoids potential state resolution conflicts where only one of the "marker" events win and we would lose the other chain history.

(emphasis added by me)

Reply:
Just talked to @adamvy and he said that Beeper clients use this as an indicator that new history has been added to the room, so I think we should keep this sending for each batch for now. Once Element properly supports this and the spec is finalized, we can revisit.